skipruntime-ts/ts/src/internals/skipruntime_impl.ts

```diff
@@ -47,21 +47,47 @@ function assertNoKeysNaN<K extends TJSON, V extends TJSON>(
 }
 export const serverResponseSuffix = "__skdb_mirror_feedback";
 
+export const sk_frozen: unique symbol = Symbol();
+
+export interface Frozen {
+  [sk_frozen]: true;
+}
+
+function sk_freeze<T extends object>(x: T): T & Frozen {
+  return Object.defineProperty(x, sk_frozen, {
+    enumerable: false,
+    writable: false,
+    value: true,
+  }) as T & Frozen;
+}
+
+function isSkFrozen(x: any): x is Frozen {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+  return sk_frozen in x && x[sk_frozen];
+}
```

Comment:
A way to help the poor type checker:
```suggestion
function isSkFrozen(x: any): x is Frozen {
  let is_frozen = false;
  if (sk_frozen in x) {
    is_frozen = x[sk_frozen];
  };
  return is_frozen;
}
```

Reply:
Thanks! `return sk_frozen in x && x[sk_frozen] === true;` works as well